crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -136,86 +132,34 @@ impl C8yMapperActor {
-                                return Ok(());
-                            }
-                        };
-
-                    let child_name = self
-                        .converter
-                        .default_device_name_from_external_id(&child_external_id);
-                    let child_topic_id = EntityTopicId::default_child_device(&child_name).unwrap();
-                    let child_device_reg_msg = EntityRegistrationMessage {
-                        topic_id: child_topic_id,
-                        external_id: Some(child_external_id),
-                        r#type: EntityType::ChildDevice,
-                        parent: None,
-                        other: json!({ "name": child_name })
-                            .as_object()
-                            .unwrap()
-                            .to_owned(),
-                    };
-                    match self
-                        .converter
-                        .register_and_convert_entity(&child_device_reg_msg)
-                    {
-                        Ok(messages) => {
-                            for message in messages {
-                                self.mqtt_publisher.send(message).await?;
-                            }
-                        }
-                        Err(err) => {
-                            error!(
-                                "Processing dynamic child device directory creation failed with {}",
-                                err
-                            );
-                        }
-                    }
-                }
-            }
             FsWatchEvent::FileCreated(path)
             | FsWatchEvent::FileDeleted(path)
-            | FsWatchEvent::Modified(path)
-            | FsWatchEvent::DirectoryDeleted(path) => {
-                match process_inotify_events(&path, file_event) {
-                    Ok(Some(discovered_ops)) => {
-                        self.mqtt_publisher
-                            .send(
-                                self.converter
-                                    .process_operation_update_message(discovered_ops),
-                            )
-                            .await?;
-                    }
-                    Ok(None) => {}
-                    Err(e) => {
-                        eprintln!("Processing inotify event failed due to {}", e);
+            | FsWatchEvent::Modified(path) => {
+                // Process inotify events only for the main device at the root operations directory
+                if path == self.converter.ops_dir || path.parent() == Some(&self.converter.ops_dir)
+                {
+                    match process_inotify_events(&path, file_event) {
+                        Ok(Some(discovered_ops)) => {
+                            self.mqtt_publisher
+                                .send(
+                                    self.converter
+                                        .process_operation_update_message(discovered_ops),
```

Comment:
There is something a bit weird/overly complex/useless here.

The `process_inotify_events` method carefully build a `DiscoverOp` with the details of the operation update,
but the `process_operation_update_message` calls `try_process_operation_update_messaDiscoverOpge` which ignores all the `message: DiscoverOp` content except for the `message.ops_dir` which is always equals to the `converter.ops_dir`.